packages/components/src/view/component.tsx

```diff
@@ -2,13 +2,25 @@
  * External dependencies
  */
 import styled from '@emotion/styled';
-import type { RefAttributes } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import type { WordPressComponent } from '../context/wordpress-component';
-import type { ViewProps } from './types';
+import type { WordPressComponentProps } from '../context';
+
+const PolymorphicDiv = styled.div``;
+
+function UnforwardedView< T extends React.ElementType = 'div' >(
+	{ as, ...restProps }: WordPressComponentProps< {}, T >,
+	ref: React.ForwardedRef< any >
+) {
+	return <PolymorphicDiv as={ as } ref={ ref } { ...restProps } />;
```

Comment:
Feels a bit off to have `PolymorphicDiv` appear in markup now, because it's an internal behavior of `View` and it's not immediately clear where it comes from. I wonder if we should keep the same naming externally so resulting markup doesn't change.

Reply:
I think "PolymorphicDiv" is a clear label that both conveys intent and is searchable, but would you prefer something prefixed like "ViewPolymorphicDiv"?

Otherwise we need to add some overhead like `const styles = { View: styled.div`` }` or whatever to get around the name collision within this file. Which is not worth it IMO, especially since the class names will be completely hashed in prod anyway.

Reply:
I'm just worried about the potential confusion that comes from the fact that the component is called `View` but it's visualized in markup as `PolymorphicDiv`. 

In practice, this could be confusing for folks who are going to be looking for a `PolymorphicDiv` component, and it could potentially be an issue for usages who were dynamically targeting selectors (which they, of course, shouldn't do) based on the presence of `-View-` in the classname. 

But if you feel this shouldn't be an issue because those internals were never official and folks should never have relied on them, then we're good.

Reply:
Oh ok, it is a non-issue for the selector usage scenario, because the Emotion class names are _completely_ hashed in prod (`<div class="css-0 e19lxcc00">`). They're impossible to use as selectors, even hackily.

For the dev mode debugging scenario, in my experience the human-readable Emotion class names are usually a soup of internal variable names, so it's only useful for identifying things when you already know what strings you're looking for.

Reply:
Thanks for confirming. Let's 🚢 !